Week1/homework/src/index.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+{
+  function fetchJSON(url, cb) {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', url);
+    xhr.responseType = 'json';
+    xhr.onload = () => {
+      if (xhr.status < 400) {
+        cb(null, xhr.response);
+      } else {
+        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        console.log('LOOK1 - GIVE ERROR');
+      }
+    };
+    xhr.onerror = () => cb(new Error('Network request failed'));
+    xhr.send();
+  }
+
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    if (options != null) {
+      Object.keys(options).forEach((key) => {
+        const value = options[key];
+        if (key === 'text') {
+          elem.innerText = value;
+        } else {
+          elem.setAttribute(key, value);
+        }
+      });
+    }
+    return elem;
+  }
+
+  function sortData(data, key) {
+    const sortedData = data.sort(function (a, b) {
+      let elem1 = a[key].toLowerCase();
+      let elem2 = b[key].toLowerCase();
+      if (elem1 < elem2) { return -1; }
+      if (elem1 > elem2) { return 1; }
+      return 0;
+    });
+    return sortedData;
+  }
+
+  function fillRepositoryList(data, root) {
+    const select = createAndAppend('select', root, { id: 'selectRepositories' });
+    const sortedData = sortData(data, 'name');
+    sortedData.forEach(elem => {
+      createAndAppend('option', select, { text: elem.name, value: elem.id });
+    });
+
+    //Listener For Repository
+    document.getElementById('selectRepositories').addEventListener("change", e => {
+      if (divRepDetails.hasChildNodes) {
+        divRepDetails.remove();
+      }
+      detailedInfoRepository();
+    });
+
+    return sortedData;
+  }
+
+  function fillUser(dataUser, root) {
+    //const sortedUser = sortData(dataUser, 'login');
+    dataUser.forEach(user => {//sortedUser
+
+      fetchJSON(user.url, (err, user1) => {
```

Comment:
As I mentioned previously, there is not need to fetch detailed information for each contributor. All the information you need is already in your `dataUser` array (which I would call just `contributors`). All these extra fetches causes the 403 error that you get.